src/gwt/src/org/rstudio/studio/client/workbench/views/source/Source.java

```diff
@@ -3109,19 +3109,26 @@ else if (type == RStudioApiRequestEvent.TYPE_DOCUMENT_OPEN)
             @Override
             public void onSuccess(final EditingTarget result)
             {
-               if (data.getRow() != -1) 
+               Command finish = () -> {
+                  // suppress attempts by the shell widget to steal focus here
+                  events_.fireEvent(new SuppressNextShellFocusEvent());
+               
+                  result.focus();
+                  result.ensureCursorVisible();
+                  
+                  JsObject response = JsObject.createJsObject();
+                  response.setString("id", result.getId());
+                  server_.rstudioApiResponse(response, new VoidServerRequestCallback());
+               };
+
+               int row = data.getRow();
+               if (row != -1) 
                {
-                  columnManager_.scrollToPosition(FilePosition.create(data.getRow(), data.getColumn()), data.getMoveCursor());
+                  FilePosition position = FilePosition.create(row, Math.max(data.getColumn(), 1));
+                  columnManager_.scrollToPosition(position, data.getMoveCursor(), finish);
```

Comment:
I'm not sure what happens if `finish` does not get called in `scrollToPosition()`.

Reply:
This isn't likely to happen, but when it does, then the `onNavigationCompleted` is never called, and so the response is not sent, which is blocking.